packaging-support/src/main/java/com/tc/classloader/PermanentEntity.java

```diff
@@ -30,8 +31,10 @@
  */
 @Target( ElementType.TYPE )
 @Retention( RetentionPolicy.RUNTIME )
+@Repeatable( Permanents.class )
 public @interface PermanentEntity {
   String type();
-  String[] names();
+  String[] names() default {};
```

Comment:
what is the use of names() now?

Reply:
Backward compatible.  You can now do either 

@PermanentEntity(type=type,names=[one,two])
public class EntityServerService

or

@PermanentEntity(type=type,names=one)
@PermanentEntity(type=type,names=two)
public class EntityServerService